pypeit/flatfield.py

```diff
@@ -190,144 +193,90 @@ def build_pixflat(self, trim=True, force=False):
-                                           spec_samp_fine=self.flatpar['spec_samp_fine'],
-                                           spec_samp_coarse=self.flatpar['spec_samp_coarse'],
-                                           spat_samp=self.flatpar['spat_samp'],
-                                           tweak_slits=self.flatpar['tweak_slits'],
-                                           tweak_slits_thresh=self.flatpar['tweak_slits_thresh'],
-                                           tweak_slits_maxfrac=self.flatpar['tweak_slits_maxfrac'],
-                                           debug=debug)
-
-            self.mspixelflat[thismask_out] = pixelflat[thismask_out]
-            self.msillumflat[thismask_out] = illumflat[thismask_out]
-            self.flat_model[thismask_out] = flat_model[thismask_out]
-
-            # Did we tweak slit boundaries? If so, update the tslits_dict and the tilts_dict
-            if self.flatpar['tweak_slits']:
-                self.tslits_dict['slit_left'][:,slit] = slit_left_out
-                self.tslits_dict['slit_righ'][:,slit] = slit_righ_out
-                self.tslits_dict['slit_left_tweak'][:,slit] = slit_left_out
-                self.tslits_dict['slit_righ_tweak'][:,slit] = slit_righ_out
-                final_tilts[thismask_out] = tilts_out[thismask_out]
-
-        # If we tweaked the slits update the tilts_dict
-        if self.flatpar['tweak_slits']:
-            self.tilts_dict['tilts'] = final_tilts
+        # Fit it
+        # NOTE: Tilts do not change and self.slits is updated
+        # internally.
+        self.fit(debug=debug)
 
         if show:
             # Global skysub is the first step in a new extraction so clear the channels here
             self.show(slits=True, wcs_match = True)
 
-        # If illumination flat fielding is turned off, set the illumflat to be None.
-        if not self.flatpar['illumflatten']:
-            msgs.warn('No illumination flat will be applied to your data (illumflatten=False).')
-            self.msillumflat = None
-
         # Return
         return self.mspixelflat, self.msillumflat
 
-    def show(self, slits=True, wcs_match=True):
+    def show(self, show_slits=True, wcs_match=True):
         """
-        Show all of the flat field products
+        Show all of the flat field products in ginga.
 
         Args:
-            slits (bool, optional):
-            wcs_match (bool, optional):
-
+            show_slits (:obj:`bool`, optional):
+                Overlay the slit edges on the flat-field images
+            wcs_match (:obj:`bool`, optional):
+                Match the WCS of the flat-field images
         """
+        ginga.connect_to_ginga(raise_err=True, allow_new=True)
+
+        # TODO: Add an option that shows the relevant stuff in a
+        # matplotlib window.
         viewer, ch = ginga.show_image(self.mspixelflat, chname='pixeflat', cuts=(0.9, 1.1),
                                       wcs_match=wcs_match, clear=True)
```

Comment:
Please show the slit boundaries on every image. This helps with understanding if the flat model is doing what it is supposed to. You can see the example in show_twodspec. Basically you just need to call show_slits after every image is shown.

Reply:
Fixed in next push.